react/features/dial-out/components/DialOutDialog.web.js

```diff
@@ -151,58 +170,33 @@ class DialOutDialog extends Component {
     /**
      * Updates the dialNumber and check for validity.
      *
-     * @param {string} dialNumber - The new value of the dialNumber property.
+     * @param {string} dialCode - The dial code value.
+     * @param {string} dialInput - The dial input value.
      * @private
      * @returns {void}
      */
-    _onDialNumberChange(dialNumber) {
-        this.setState({
-            dialNumber
-        });
+    _onDialNumberChange(dialCode, dialInput) {
+        // We remove all starting zeros from the dial input before attaching it
+        // to the country code.
+        const formattedDialInput = dialInput.replace(/^(0+)/, '');
 
-        if (dialNumber && dialNumber.length > 0) {
-            this.props._checkDialNumber(dialNumber);
-        }
-    }
-}
+        const dialNumber
+            = dialCode
+            ? `${dialCode}${formattedDialInput}`
+            : formattedDialInput;
 
-/**
- * Maps parts of Redux actions to component props.
- *
- * @param {Function} dispatch - Redux action dispatcher.
- * @returns {{
- *     _cancel: Function,
- *     _dial: Function,
- *     _checkDialNumber: Function
- * }}
- * @private
- */
-function _mapDispatchToProps(dispatch: Function): Object {
-    return {
-        _cancel() {
-            dispatch(cancel());
-        },
+        const formattedNumber = this._formatDialNumber(dialNumber);
 
-        /**
-         * Dispatches a (redux) action to set the default toolbar buttons.
-         *
-         * @param {string} dialNumber - The number to dial.
-         * @returns {Object} Dispatched action.
-         */
-        _dial(dialNumber) {
-            dispatch(dial(dialNumber));
-        },
+        this.setState({
```

Comment:
Why not create `const isDialNumberEmpty = !formattedDialInput || formattedDialInput.length === 0` and use it for the state and the check bellow instead of calculating it twice?